auth-agents-common/src/main/java/org/apache/atlas/policytransformer/CachePolicyTransformerImpl.java

```diff
@@ -146,26 +148,27 @@ public ServicePolicies getPolicies(String serviceName, String pluginId, Long las
             servicePolicies.setPolicyUpdateTime(new Date());
 
             if (service != null) {
-                List<RangerPolicy> policies = getServicePolicies(service);
+                List<RangerPolicy> allPolicies = getServicePolicies(service);
                 servicePolicies.setServiceName(serviceName);
-                servicePolicies.setPolicies(policies);
+                //servicePolicies.setPolicies(policies);
                 servicePolicies.setServiceId(service.getGuid());
 
                 String serviceDefName = String.format(RESOURCE_SERVICE_DEF_PATTERN, serviceName);
                 servicePolicies.setServiceDef(getResourceAsObject(serviceDefName, RangerServiceDef.class));
 
 
                 //Process tag based policies
-                String tagServiceName = (String) service.getAttribute("tagService");
+                String tagServiceName = (String) service.getAttribute(ATTR_SERVICE_TAG_SERVICE);
                 if (StringUtils.isNotEmpty(tagServiceName)) {
                     AtlasEntityHeader tagService = getServiceEntity(tagServiceName);
 
                     if (tagService != null) {
-                        policies = getServicePolicies(tagService);
```

Comment:
Earlier when you were updating policies attribute here, wasn't this updating the policies set at line 151 above?

Reply:
no, here we were assigning completely new List object to existing reference `policies`, this was not updating the policies set at line 151.